tests/ga/test_extension.py

```diff
@@ -605,6 +605,50 @@ def test_ext_handler(self, *args):
 
         self._assert_no_handler_status(protocol.report_vm_status)
 
+    def test_it_should_zip_waagent_status_when_incarnation_changes(self, *args):
+        """
+        This test checks when the incarnation changes the waagnet_status file for the previous incarnation
+        is added into the history folder for the previous incarnation and gets zipped
+        """
+        temp_files = [
+            'ExtensionsConfig.1.xml',
+            'GoalState.1.xml',
+            'OSTCExtensions.ExampleHandlerLinux.1.manifest.xml',
+            'waagent_status.1.json'
+        ]
+
+        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)  # pylint: disable=no-value-for-parameter
+
+        exthandlers_handler.run()
+        exthandlers_handler.report_ext_handlers_status()
+
+        # Updating incarnation to 2 , hence the history folder should have waaagent_status.1.json added under
+        # incarnation 1
+        test_data.set_incarnation(2)
+        protocol.update_goal_state()
+
+        test_subject = StateArchiver(self.tmp_dir)
+        test_subject.archive()
+
+        timestamp_zips = os.listdir(os.path.join(self.tmp_dir, "history"))
+        self.assertEqual(1, len(os.listdir(os.path.join(self.tmp_dir, "history"))))
```

Comment:
should we use timestamp_zips instead of os.listdir(os.path.join(self.tmp_dir, "history")?

Reply:
Please add an appropriuate message  to the assertion for the case when it fails

Reply:
Addressed in next commit